sc-memory/sc-memory/sc_agent.tpp

```diff
@@ -0,0 +1,620 @@
+      {ScKeynodes::sc_event_before_erase_edge, {GetIteratorForEventTripleWithEdge, 2u}},
+      {ScKeynodes::sc_event_after_generate_connector,
+       {GetIteratorForEventTripleWithConnectorWithOutgoingDirection, 2u}},
+      {ScKeynodes::sc_event_before_erase_connector, {GetIteratorForEventTripleWithConnectorWithOutgoingDirection, 2u}}};
+
+  ScAddr const & eventClassAddr = event.GetEventClass();
+  auto const & iteratorIt = eventToEventTripleIterators.find(eventClassAddr);
+  if (iteratorIt == eventToEventTripleIterators.cend())
+  {
+    SC_LOG_WARNING(
+        "Event class for agent class `"
+        << this->GetName()
+        << "` is unknown. It is impossible to check initiation condition (or result condition) template.");
+    return ScTemplate();
+  }
+
+  auto [getIteratorForEventTriple, otherElementPosition] = iteratorIt->second;
+  ScIterator5Ptr eventTripleIterator = getIteratorForEventTriple();
+  bool checkTemplateParamsIsGenerated = false;
+  ScTemplateParams checkTemplateParams;
+  if (eventTripleIterator->IsValid())
+    checkTemplateParamsIsGenerated = GenerateCheckTemplateParams(
+        checkTemplateAddr,
+        eventSubscriptionElementAddr,
+        otherElementAddr,
+        otherElementPosition,
+        eventTripleIterator,
+        checkTemplateParams);
+
+  if (checkTemplateParamsIsGenerated
+      && (eventClassAddr == ScKeynodes::sc_event_after_generate_connector
+          || eventClassAddr == ScKeynodes::sc_event_before_erase_connector))
+  {
+    eventTripleIterator = GetIteratorForEventTripleWithConnectorWithIncomingDirection();
+    if (eventTripleIterator->IsValid())
+    {
+      otherElementPosition = 0u;
+      GenerateCheckTemplateParams(
+          checkTemplateAddr,
+          eventSubscriptionElementAddr,
+          otherElementAddr,
+          otherElementPosition,
+          eventTripleIterator,
+          checkTemplateParams);
+    }
+  }
+
+  ScTemplate checkTemplate;
+  this->m_context.HelperBuildTemplate(checkTemplate, checkTemplateAddr, checkTemplateParams);
+  return checkTemplate;
+}
+
+template <class TScEvent, class TScContext>
+bool ScAgent<TScEvent, TScContext>::GenerateCheckTemplateParams(
+    ScAddr const & checkTemplateAddr,
+    ScAddr const & eventSubscriptionElementAddr,
+    ScAddr const & otherElementAddr,
+    size_t otherElementPosition,
+    ScIterator5Ptr const eventTripleIterator,
+    ScTemplateParams & checkTemplateParams) noexcept
+{
+  ScTemplateParams params;
+  ScIterator3Ptr const subscriptionElementIterator =
+      this->m_context.Iterator3(checkTemplateAddr, ScType::EdgeAccessConstPosPerm, eventSubscriptionElementAddr);
+  if (subscriptionElementIterator->Next())
+  {
+    if (eventTripleIterator->Next())
+    {
+      ScAddr const & otherVarAddr = eventTripleIterator->Get(otherElementPosition);
+      if (this->m_context.GetElementType(otherVarAddr).IsVar() && otherVarAddr != otherElementAddr)
+        params.Add(otherElementPosition, otherElementAddr);
```

Comment:
this line uses 2 as ScAddr hash, that's why clion says Single-argument constructors must be marked explicit to avoid unintentional implicit conversions

Reply:
`otherElementPosition` has type `size_t`. Compiler will use this type. If `otherElementPosition` has `auto` type, then compiler can't choose correct method

Reply:
Fixed